File: scanner/corpora/go/positive/ed25519.go
package main

import (
	"crypto/ed25519"
	"fmt"
)

func ed25519Sign(message []byte) ([]byte, []byte) {
	publicKey, privateKey, _ := ed25519.GenerateKey(nil)
	signature := ed25519.Sign(privateKey, message)
	return signature, publicKey
}

func ed25519Verify(message, signature, publicKey []byte) bool {
	return ed25519.Verify(publicKey, message, signature)
}

func main() {
	message := []byte("Message to sign")
	sig, pub := ed25519Sign(message)
	valid := ed25519Verify(message, sig, pub)
	fmt.Printf("Signature valid: %v\n", valid)
}

File: scanner/corpora/python/positive/ed25519.py
"""Ed25519 digital signature using the cryptography library."""
from cryptography.hazmat.primitives.asymmetric.ed25519 import (
    Ed25519PrivateKey
)

private_key = Ed25519PrivateKey.generate()
public_key = private_key.public_key()
message = b"Message to sign"
signature = private_key.sign(message)
public_key.verify(signature, message)
print("Ed25519 signature verified")
